src/sst/core/serialization/serializer.h

```diff
@@ -290,8 +260,31 @@ class serializer
     // Used for unpacking and mapping
     std::map<uintptr_t, uintptr_t> ser_pointer_map;
     uintptr_t                      split_key;
-    std::stack<std::string>        map_name;
-};
+    const ObjectMapContext*        mapContext = nullptr;
+    friend class ObjectMapContext;
+}; // class serializer
+
+/**
+   ObjectMap context which is saved in a virtual stack when name or other information changes.
+   When ObjectMapContext is destroyed, the serializer goes back to the previous ObjectMapContext.
+ */
+
+class ObjectMapContext
```

Comment:
This seems like a good replacement for the name stack.  However, the plan is to still also pass an integer options value to each serialize_impl call as a bitwise map of options.  These options are needed in SIZE, PACK and UNPACK mode and we're not willing to create a context object in those modes as they are used in the serialization for synchronization fast path.

Reply:
I think the flags should be a mostly-opaque type, such as `SerFlag`. I would not want `uint32_t` or `uint64_t` to be exposed in the API as the type of the flags. There should be a separate `using` (like`typedef`) or `enum class` for the flags, using whatever integer underlying type wanted. The names should be prevented from collision with other names, such as putting them in an `enum class` or an `enum` inside the public section of a class, or a namespace with a short name like `Ser`.

Like how [`RevFlag` is implemented](https://github.com/tactcomplabs/rev/blob/devel/include/RevFlag.h).

Reply:
That's how I've got them implemented.